src/uxbox/ui/dashboard/projects.cljs

```diff
@@ -149,30 +150,72 @@
 ;; Menu
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-;; (def ^:static menu-l
-;;   (as-> (l/select-keys [:projects]) $
-;;     (l/focus-atom $ s/state)))
-
-(rum/defc project-sort-selector < rum/reactive
-  [sort-order]
-  nil)
-  ;; (let [sort-name (get project-orderings (rum/react sort-order))]
-  ;;   [:select.input-select
-  ;;    {:on-change #(reset! sort-order (name->order (.-value (.-target %))))
-  ;;     :value sort-name}
-  ;;    (for [order (keys project-orderings)
-  ;;          :let [name (get project-orderings order)]]
-  ;;      [:option {:key name} name])]))
+(def ^:static menu-l
+  (as-> (l/select-keys [:projects-by-id]) $
+    (l/focus-atom $ s/state)))
+
+(def ^:static project-ordering-l
+  (as-> (l/in [:dashboard :project-order]) $
+    (l/focus-atom $ s/state)))
+
+(def ^:static project-filtering-l
+  (as-> (l/in [:dashboard :project-filter]) $
+    (l/focus-atom $ s/state)))
+
+(defn project-sort-render
+  []
+  (let [ordering (rum/react project-ordering-l)]
+    (html
+     [:div
+       [:span (tr "ds.project-ordering")]
+       [:select.input-select
+        {:on-change #(rs/emit! (dd/set-project-ordering (keyword (.-value (.-target %)))))
+         :value (name ordering)}
+        (for [option (keys +ordering-options+)
+              :let [option-id (get +ordering-options+ option)
+                    option-value (name option)
+                    option-text (tr option-id)]]
+          [:option
+           {:key option-id
+            :value option-value}
+           option-text])]])))
+
+(def project-sorting
+  (mx/component
+   {:render project-sort-render
+    :name "project-sort-order"
+    :mixins [rum/reactive]}))
+
+(defn project-search-render
+  []
+  (html
+    [:form
+     [:input
+      {:type "text"
+       :on-change #(rs/emit! (dd/set-project-filtering (.-value (.-target %))))
```

Comment:
Please, declare the functions in `(let... ` just like this https://github.com/uxbox/uxbox/blob/master/src/uxbox/ui/workspace/colorpalette.cljs#L55

This makes better and visual separation on logic and data :P